dj/sql/functions.py

```diff
@@ -98,6 +98,11 @@ class Function:  # pylint: disable=too-few-public-methods
     def infer_type(*args: Any) -> ColumnType:
         raise NotImplementedError("Subclass MUST implement infer_type")
 
+    @staticmethod
+    @abc.abstractmethod
+    def infer_type_from_types(*args: Any) -> ColumnType:
+        raise NotImplementedError("Subclass MUST implement infer_type_from_types")
```

Comment:
Since this is an `abstractmethod`, doesn't it already raise an error if the function isn't implemented in a child class? I noticed the you have full coverage on the PR though so this line must be getting hit...I'm missing something aren't I? What would happen if you just had an empty function here with a docstring?:
```py
    @staticmethod
    @abc.abstractmethod
    def infer_type_from_types(*args: Any) -> ColumnType:
        """
        Infer the return type of a function given the types of its arguments
        """
```

Reply:
Actually just saw that the implementations don't have the same args part of the function signature. Maybe this shouldn't be an `abstractmethod` then and you can just drop the decorator. That should let you remove all of those `# type: ignore` comments on the implementations.

Reply:
I just mimicked the existing `infer_type` creating this method to fit with the ast inference